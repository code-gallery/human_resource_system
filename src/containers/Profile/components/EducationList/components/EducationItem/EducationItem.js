import React from 'react'
import PropTypes from 'prop-types'
import EducationItemView from '../EducationItemView'
import EducationItemEdit from '../EducationItemEdit'

class EducationItem extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      edit: false
    }
  }

  componentWillReceiveProps = (nextProps) => {
    if (this.state.edit && this.props.editMode && !nextProps.editMode) {
      this.toggleEdit()
    }
  }

  toggleEdit = () => {
    this.setState({
      edit: !this.state.edit
    })
    this.props.toggleBlockEditing()
  }

  render() {
    const { edit } = this.state
    const { deleteEducation } = this.props
    return (
      <div>
        <EducationItemView
          {...this.props}
          toggleEdit={this.toggleEdit}
        />
        {
          edit &&
          <EducationItemEdit
            {...this.props}
            toggleEdit={this.toggleEdit}
            edit={true}
            deleteEducation={deleteEducation}
          />
        }
      </div>
    )
  }
}

EducationItem.propTypes = {
  toggleBlockEditing: PropTypes.func.isRequired,
  deleteEducation: PropTypes.func.isRequired,
  saveEntity: PropTypes.func.isRequired,
  editMode: PropTypes.bool
}

export default EducationItem
