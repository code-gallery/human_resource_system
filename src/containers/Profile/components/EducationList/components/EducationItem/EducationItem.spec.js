import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import EducationItem from './EducationItem'

describe('containers/Profile/components/.../EducationItem', () => {
  const commonProps = {
    created_at: '2017-12-13 15:17:02',
    degree: 'MSc',
    start_date: '2004-09-01',
    end_date: '2005-08-01',
    entity_type: 'educations',
    grade: 'Pass',
    org: {
      id: 953,
      name: 'Oxford Brookes University',
      logo_image: '//appii.s3.eu-west-2.amazonaws.com/organisation_logos/oxford-brookes-university1509337248159.jpg'
    },
    verified_status: 'not_verified',
    type: 'higher',
    visibility: 'registered',
    toggleBlockEditing: jest.fn(),
    deleteEducation: jest.fn(),
    saveEntity: jest.fn()
  }

  describe('when editMode is false', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        editMode: false
      }
      const tree = renderer.create(
        <EducationItem {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when editMode is true', () => {
    const props = {
      ...commonProps,
      editMode: true
    }

    it('renders correctly', () => {
      const tree = renderer.create(
        <EducationItem {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })

    describe('componentWillReceiveProps', () => {
      it('calls props.toggleBlockEditing correctly', () => {
        const wrapper = shallow(
          <EducationItem {...props} />
        )
        expect(props.toggleBlockEditing).not.toHaveBeenCalled()
        wrapper.setState({ edit: true }, function() {
          wrapper.setProps({
            editMode: false
          })
        })
        expect(props.toggleBlockEditing).toHaveBeenCalled()
        expect(wrapper.state('edit')).toEqual(false)
      })
    })
  })
})
