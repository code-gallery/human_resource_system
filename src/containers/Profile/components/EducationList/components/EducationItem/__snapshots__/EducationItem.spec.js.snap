// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/.../EducationItem when editMode is false renders correctly 1`] = `
<div>
  <div
    className="profile-experience-block profile-unverified-status"
  >
    <div
      className="relative"
    >
      <div
        className="Avatar Avatar-white Avatar-size-109"
      >
        <img
          alt="logo"
          src="appii-placeholder.png"
        />
      </div>
      <p
        className="visible-xs visible-sm"
      >
        <i
          className="profile-status-ico tooltip-position-top tooltip-movable tooltip-dark"
          data-tooltip="Not verified"
        >
           
        </i>
      </p>
    </div>
    <div
      className="profile-experience-info"
    >
      <i
        className="profile-status-ico tooltip-position-top tooltip-movable tooltip-dark hidden-xs hidden-sm"
        data-tooltip="Not verified"
      >
         
      </i>
      <p>
        Oxford Brookes University
      </p>
      <span>
        MSc
        
      </span>
      <em>
        2004
         - 
        2005
      </em>
    </div>
  </div>
</div>
`;

exports[`containers/Profile/components/.../EducationItem when editMode is true renders correctly 1`] = `
<div>
  <div
    className="profile-experience-block profile-unverified-status"
  >
    <div
      className="profile-experience-btns"
    >
      <button
        className="edit-btn"
        onClick={[Function]}
      >
         
      </button>
    </div>
    <div
      className="relative"
    >
      <div
        className="Avatar Avatar-white Avatar-size-109"
      >
        <img
          alt="logo"
          src="appii-placeholder.png"
        />
      </div>
      <p
        className="visible-xs visible-sm"
      >
        <i
          className="profile-status-ico tooltip-position-top tooltip-movable tooltip-dark"
          data-tooltip="Not verified"
        >
           
        </i>
      </p>
    </div>
    <div
      className="profile-experience-info"
    >
      <i
        className="profile-status-ico tooltip-position-top tooltip-movable tooltip-dark hidden-xs hidden-sm"
        data-tooltip="Not verified"
      >
         
      </i>
      <p>
        Oxford Brookes University
      </p>
      <span>
        MSc
        
      </span>
      <em>
        2004
         - 
        2005
      </em>
    </div>
  </div>
</div>
`;
